packages/contracts/contracts/MarketRegistry.sol

```diff
@@ -545,16 +560,25 @@ contract MarketRegistry is
      * @notice Sets the duration of new loans for this market before they turn delinquent.
      * @notice Changing this value does not change the terms of existing loans for this market.
      * @param _marketId The ID of a market.
+     * @param _paymentCycleType Cycle type (seconds or monthly)
      * @param _duration Delinquency duration for new loans
      */
-    function setPaymentCycleDuration(uint256 _marketId, uint32 _duration)
+    function setPaymentCycle(uint256 _marketId, PaymentCycleType _paymentCycleType, uint32 _duration)
         public
         ownsMarket(_marketId)
     {
-        if (_duration != markets[_marketId].paymentCycleDuration) {
-            markets[_marketId].paymentCycleDuration = _duration;
+        require(
+            (_paymentCycleType == PaymentCycleType.Seconds) ||
+            (_paymentCycleType == PaymentCycleType.Monthly && _duration == 0),
+            "Monthly payment cycle value must be 0"
+        );
+        Marketplace storage market = markets[_marketId];
+        uint32 duration = _paymentCycleType == PaymentCycleType.Seconds ? _duration : 30 days;
+        if (_paymentCycleType != market.paymentCycleType || duration != market.paymentCycleDuration) {
+            markets[_marketId].paymentCycleType = _paymentCycleType;
+            markets[_marketId].paymentCycleDuration = duration;
```

Comment:
what is the purpose of setting this to '30 days' in the monthly type?  I dont think that '30 days' value is even used anywhere.